spec/requests/api/v0/tags_spec.rb

```diff
@@ -45,6 +45,25 @@
       expect(response.parsed_body.map { |t| t["id"] }).to match_array(tag_ids)
     end
 
+    it "finds tags from array of tag_names" do
+      tags = create_list(:tag, 10, taggings_count: 10)
+      tag_names = tags.sample(4).map(&:name)
+
+      get api_tags_path, params: { tag_names: tag_names }
+
+      expect(response.parsed_body.map { |t| t["name"] }).to match_array(tag_names)
+    end
+
+    it "finds tags from array of tag_ids if both tag_ids and tag_names is passed" do
+      tags = create_list(:tag, 10, taggings_count: 10)
+      tag_names = tags.sample(2).map(&:name)
+      tag_ids = tags.sample(4).map(&:id)
+
+      get api_tags_path, params: { tag_names: tag_names, tag_ids: tag_ids }
+
+      expect(response.parsed_body.map { |t| t["id"] }).to match_array(tag_ids)
```

Comment:
I'm kind of surprised this passed 😅 I thought the `match_array` method ensured it had everything. in this case, it appears that you have an array of ids with a length of 6. but, the `tag_ids` variable should only contain 4 tag ids. maybe it does partial matching 🤷 -- I'll have to look into that. 

anyhow, maybe we should check that response contains both? maybe something as simple as what's below will suffice?


```
expect(response.parsed_body.map { |t| t["name"] }).to match_array(tag_names)
expect(response.parsed_body.size).to eq 6
```

Reply:
Hmm I think there's a misunderstanding - the test is expected to only return 4 tags (because when both `tag_ids` and `tag_names` are passed, only `tag_ids` will be used). And so I'm expecting to only get back the 4 tags that I passed the IDs for on line 62.

So I think it's right?

Reply:
@aitchiss your rspec looks right.

Reply:
whoops! you're right. I read the if statement in the controller wrong! my bad!